Sources/SwiftParser/Declarations.swift

```diff
@@ -1357,7 +1357,13 @@ extension Parser {
           defaultArgument = nil
         }
 
-        let trailingComma = self.consume(if: .comma)
+        var trailingComma = self.consume(if: .comma)
+        if trailingComma == nil {
+          var lookahead = self.lookahead()
+          if lookahead.consume(if: { $0.canBeArgumentLabel }, followedBy: { $0.tokenKind == .colon }) != nil {
```

Comment:
Hmm, I’m not sure if this is the right approach because it break for anything but the most simple arguments. E.g. we wouldn’t enter this recovery case if the argument has modifiers or a `secondName`. I think you would need to implement a little more sophisticated logic in `Lookahead` and call that from here.

Also adding test a test case for `func foo(first second: Int third fourth: Int)` would be good.